src/libraries/System.Net.Security/tests/InternalUnitTests/MD4Tests.cs

```diff
@@ -0,0 +1,84 @@
+        }
+
+        // // MD4("a") = bde52cb31de33e46245e05fbdbd6fb24
+        [Fact]
+        public void TryEncrypt_SingleLetter()
+        {
+            ReadOnlySpan<byte> input = new ReadOnlySpan<byte>(Encoding.Default.GetBytes("a"));
+            ReadOnlySpan<byte> expected = new byte[] { 0xbd, 0xe5, 0x2c, 0xb3, 0x1d, 0xe3, 0x3e, 0x46, 0x24, 0x5e, 0x05, 0xfb, 0xdb, 0xd6, 0xfb, 0x24 };
+            Verify(input, expected);
+        }
+
+        // MD4("abc") = a448017aaf21d8525fc10ae87aa6729d
+        [Fact]
+        public void TryEncrypt_ThreeLetters()
+        {
+            ReadOnlySpan<byte> input = new ReadOnlySpan<byte>(Encoding.Default.GetBytes("abc"));
+            ReadOnlySpan<byte> expected = new byte[] { 0xa4, 0x48, 0x01, 0x7a, 0xaf, 0x21, 0xd8, 0x52, 0x5f, 0xc1, 0x0a, 0xe8, 0x7a, 0xa6, 0x72, 0x9d };
+            Verify(input, expected);
+        }
+
+        // MD4("message digest") = d9130a8164549fe818874806e1c7014b
+        [Fact]
+        public void TryEncrypt_Phrase()
+        {
+            ReadOnlySpan<byte> input = new ReadOnlySpan<byte>(Encoding.Default.GetBytes("message digest"));
+            ReadOnlySpan<byte> expected = new byte[] { 0xd9, 0x13, 0x0a, 0x81, 0x64, 0x54, 0x9f, 0xe8, 0x18, 0x87, 0x48, 0x06, 0xe1, 0xc7, 0x01, 0x4b };
+            Verify(input, expected);
+        }
+
+        // MD4("abcdefghijklmnopqrstuvwxyz") = d79e1c308aa5bbcdeea8ed63df412da9
+        [Fact]
+        public void TryEncrypt_AlphabetInLowercase()
+        {
+            ReadOnlySpan<byte> input = new ReadOnlySpan<byte>(Encoding.Default.GetBytes("abcdefghijklmnopqrstuvwxyz"));
+            ReadOnlySpan<byte> expected = new byte[] { 0xd7, 0x9e, 0x1c, 0x30, 0x8a, 0xa5, 0xbb, 0xcd, 0xee, 0xa8, 0xed, 0x63, 0xdf, 0x41, 0x2d, 0xa9 };
+            Verify(input, expected);
+        }
+
+        // MD4("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789") =	043f8582f241db351ce627e153e7f0e4
+        [Fact]
+        public void TryEncrypt_AlphabetInUpperLowerCasesAndNumbers()
+        {
+            ReadOnlySpan<byte> input = new ReadOnlySpan<byte>((Encoding.Default.GetBytes("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789")));
+            ReadOnlySpan<byte> expected = new byte[] { 0x04, 0x3f, 0x85, 0x82, 0xf2, 0x41, 0xdb, 0x35, 0x1c, 0xe6, 0x27, 0xe1, 0x53, 0xe7, 0xf0, 0xe4 };
+            Verify(input, expected);
+        }
+
+        // MD4("12345678901234567890123456789012345678901234567890123456789012345678901234567890") = e33b4ddc9c38f2199c3e7b164fcc0536
+        [Fact]
+        public void TryEncrypt_RepeatedSequenceOfNumbers()
+        {
+            ReadOnlySpan<byte> input = new ReadOnlySpan<byte>(Encoding.Default.GetBytes("12345678901234567890123456789012345678901234567890123456789012345678901234567890"));
+            ReadOnlySpan<byte> expected = new byte[] { 0xe3, 0x3b, 0x4d, 0xdc, 0x9c, 0x38, 0xf2, 0x19, 0x9c, 0x3e, 0x7b, 0x16, 0x4f, 0xcc, 0x05, 0x36 };
+            Verify(input, expected);
+        }
+
```

Comment:
These tests are probably fine.  If you want some more interesting cases:

// 55 bytes (padLen == 56 - 55 => 1)
ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz012 => 14fdf2056bf88b3491c385d8ac4f48e6

// 56 bytes (padLen == 120 - 56 => 64)
ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123 => db837dbb6098a50a2d3974bc1cc76133

And then both of those padded out by 64 more bytes.

Lengths 63, 64, 65, 127, 128, 129 are probably also interesting, since they hit the mod boundaries.

Values produced by https://emn178.github.io/online-tools/md4.html, which agreed on 12345678901234567890123456789012345678901234567890123456789012345678901234567890 and abc